modules/ocf_backups/manifests/offsite.pp

```diff
@@ -19,5 +28,25 @@
     '/opt/share/backups/boto.cfg':
       source => 'puppet:///private/boto.cfg',
       mode   => '0600';
+
+    # Box.com credentials and API id/secret
+    '/opt/share/backups/box-creds.json':
+      source => 'puppet:///private/box-creds.json',
+      mode   => '0600';
+
+    # This changes each time a backup is made, so it is not stored on the
+    # puppetmaster and instead is just a local file on the backups server.
+    '/opt/share/backups/box-refresh-token':
+      mode => '0600';
```

Comment:
Can you separate token refreshing into a separate script and run it in a cronjob on the puppetmaster?

Alternatively:
* MySQL would works but is a little heavy for storing one value
* NFS would work but we don't mount NFS on hal currently, and it's really bad for secrets (it's not encrypted at all)